railties/lib/rails/application/configuration.rb

```diff
@@ -220,6 +220,21 @@ def load_defaults(target_version)
           if respond_to?(:action_mailer)
             action_mailer.smtp_timeout = 5
           end
+
+          if respond_to?(:active_storage)
+            active_storage.video_preview_arguments = [
+              "-vf",
+              # Select the first video frame, plus keyframes and frames
+              # that meet the scene change threshold.
+              'select=eq(n\,0)+eq(key\,1)+gt(scene\,0.015),' +
+              # Loop the first 1-2 selected frames in case we were only
+              # able to select 1 frame, then drop the first looped frame.
+              # This lets us use the first video frame as a fallback.
+              "loop=loop=-1:size=2,trim=start_frame=1",
+              "-frames:v", "1",
+              "-f", "image2", "-"
```

Comment:
This long literal and its longer explanation feel a little out of place in this file.  Maybe there is a good place to extract these to, as a constant? :thinking:  If not, perhaps extract the explanation to the Configuration Guide?  (It might be helpful there, anyway.)


Also, I've been wanting to tweak the explanation's wording for a little while :hourglass::

```suggestion
              "-vf",
              # Select the first video frame, plus keyframes, plus frames
              # that meet the scene change threshold.
              'select=eq(n\,0)+eq(key\,1)+gt(scene\,0.015),' +
              # To use the first video frame as a fallback when no other
              # frames meet the criteria, loop the first (one or) two
              # selected frames, then drop the first looped frame.
              "loop=loop=-1:size=2,trim=start_frame=1",
              "-frames:v", "1",
              "-f", "image2", "-"
```

Reply:
This sounds reasonable. Perhaps @pixeltrix could chime in on this one too? I'm happy to move it. Having it as a single string would also make the inline comments less useful here.

Reply:
Maybe the explanation could also go in the upgrading guide?

https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-6-1-to-rails-7-0

Reply:
To clarify, I think the comments would be better in the configuration guide as even if you're looking at the Active Storage code you won't see them as this is Railties.

Reply:
I've written a short spiel in the upgrade guide and have outlined the technique in the configurations guide.